packages/shop/src/hooks/use-search/useSearch.ts

```diff
@@ -0,0 +1,54 @@
+import { useInfiniteQuery } from 'react-query';
+
+import {
+  SearchApiRequestBody,
+  SearchApiSuccessResponse,
+  SearchApiErrorResponse,
+} from '../../pages/api/search/post';
+
+const QUERY_KEY = 'search';
+
+type Options = {
+  searchWord: string;
+  staleTime?: number;
+};
+
+export const useSearch = (options: Options) => {
+  const { searchWord, staleTime } = options;
+
+  return useInfiniteQuery<SearchApiSuccessResponse, SearchApiErrorResponse>(
+    [QUERY_KEY, searchWord],
+    async ({ pageParam = 1 }) => {
+      const body: SearchApiRequestBody = {
+        searchWord: searchWord.trim(),
+        pageIndex: pageParam,
+      };
+
+      const response = await fetch(`/api/search`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body),
+      });
+
+      if (!response.ok) {
+        throw await response.json();
+      }
+
+      return response.json();
+    },
+    {
+      enabled: !!searchWord,
```

Comment:
nit
```suggestion
      enabled: !!(searchWord.trim()),
```

Reply:
Fixed in https://github.com/moloco-rmp/demo-store/pull/10/commits/41b83b1f548938c88d9cc2859228f1971506a08a.